### PR TITLE
feat: catch nix errors

### PR DESCRIPTION
**Pull request by mkenigs**
- Add LockFlakeException to catch any exceptions thrown while calling nix::flake::lockFlake
- Add NixEvalException to catch nix::EvalError
- Catch any other nix::Error in main()
- Minor fix: use whatString instead of what for FloxException in main, as it will contain more information

### CODE REVIEW

File: src/registry.cc [hunk 1]
```diff
@@ -147,20 +147,28 @@ FloxFlakeInput::getSubtrees()
...
+                  this->enabledSubtrees = std::vector<Subtree> { ST_PACKAGES };
+                }
+              else if ( root->maybeGetAttr( "legacyPackages" ) != nullptr )
+                {
+                  this->enabledSubtrees = std::vector<Subtree> { ST_LEGACY };
+                }
+              else { this->enabledSubtrees = std::vector<Subtree> {}; }
             }
-          else if ( root->maybeGetAttr( "packages" ) != nullptr )
+          catch ( const nix::EvalError &err )
```
Comment (mkenigs) [inline src/registry.cc:167]: @aakropotkin I was just guessing this is the right error to catch here, do you know if that's correct?

**Comment by mkenigs** [comment]
@aakropotkin I didn't catch IFD build errors, if we need to do that could you point me towards where that happens/what error gets thrown?

**Review by zmitchell — APPROVED** [review]

**Comment by zmitchell** [comment]
I've approved and enabled auto-merge. If you need to make changes feel free to open a separate PR and I'll fast-track it.

File: include/flox/core/exceptions.hh [hunk 2]
```diff
@@ -33,31 +34,37 @@ enum error_category {
   /** Generic exception emitted by `flox` routines. */
   EC_FLOX_EXCEPTION = 100,
   /** A command line argument is invalid. */
-  EC_INVALID_ARG = 101,
+  EC_INVALID_ARG,
```
[the remark this answers is not in the document]
Comment (mkenigs) [reply include/flox/core/exceptions.hh:37]: Yep! "For any other enumerator whose definition does not have an initializer, the associated value is the value of the previous enumerator plus one"

File: src/registry.cc (hunk 1 above)
Comment (aakropotkin) [reply src/registry.cc:167]: Yeah you got the right one.
This is the underlying `throw` : https://github.com/flox/pkgdb/blob/main/src/pkgdb/write.cc#L446

**Pull request merged**